### DISCUSSION TITLE
`canvas` scaling using a `q-splitter`

### DISCUSSION THREAD

**Discussion by srothst1** [quasarframework/quasar#14027] [General - Components / Directives / etc]
Hi Quasar community :wave: I am having an issue with `canvas` sizing when using a `q-splitter`. The following image showcases what I have so far:
![Screenshot from 2022-07-22 08-56-19](https://user-images.githubusercontent.com/39531367/180477804-97713ce4-92da-48e8-898c-e706e969f4a7.png)

I would like the `canvas` objects to fill the entire left (`v-slot:before`) and right (`v-slot:after`) sides of the `q-splitter`. In addition, I do not want a horizontal or vertical scroll bar to appear when I am resizing the UI elements or the window.

I have [created a codepen demo](https://codepen.io/srothst1/pen/bGvRWmQ?editors=1010) that showcases my current progress. Please let me know if you have any suggestions.

```.HTML
<div id="q-app">
  <div>
    <q-splitter v-model="splitterModel">
 
      <template v-slot:before>
        <div class="q-pa-md">
          <canvas ref="multilogPlaybackAppCanvas1" style="border:1px solid #000000;"/>
        </div>
      </template>

      <template v-slot:separator>
        <q-avatar color="primary" text-color="white" size="40px" icon="drag_indicator"></q-avatar>
      </template>

      <template v-slot:after>
        <div class="q-pa-md">
          <canvas ref="multilogPlaybackAppCanvas1" style="border:1px solid #000000;"/>
        </div>
      </template>

    </q-splitter>
  </div>
</div>
```

**Comment by srothst1** [comment]
Hey everyone :wave: Any updates on this thread?
I have done some additional research in the interum. It seems like the `splitpanes` API may be a viable solution: https://github.com/antoniandre/splitpanes

**Reply by metalsadman** [reply]
Make your canvases width styles reactive property relative to the width of your splitter, check its api I believe there's an event that supplies those values.

**Reply by srothst1** [reply]
Hi @metalsadman. Are you able to provide the `splitpanes` documentation for that object? I was not able to find what you are referencing.

Ideally, I would be able to meet [my needs](https://github.com/quasarframework/quasar/discussions/14027#discussion-4238458) using only the base Quasar API.
